daemon/logger/factory/factory.go

```diff
@@ -0,0 +1,57 @@
+package factory
+
+import (
+	"fmt"
+	"sync"
+
+	logger "github.com/docker/docker/daemon/logger"
+)
+
+// Creator is a method that builds a logging driver instance with given context
+type Creator func(LoggingContext) (logger.Logger, error)
+
+// LoggingContext provides enough information for a logging driver to do its function
+type LoggingContext struct {
```

Comment:
@LK4D4 looks like opts are `map[string]string`? If that's the case we can add `Opts map[string]string` here?